### PR TITLE
Move the golden-test runner from the GH action yaml to a .sh file

### PR DESCRIPTION
**Pull request by drevell**
Before, you couldn't run this script manually, because it's embedded in the GitHub action YAML file. Now you can run it at will on your local machine.

### CODE REVIEW

File: run_golden_tests.sh [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+# This script is intended to be invoked in a github action. It will run all the
+# the golden tests that it can find.
+
+set -eEuo pipefail
+
+exit_status=0
+for template_dir in $(find . -name 'testdata' | grep -v '/.git/' | xargs dirname) ; do
+    echo "Running golden tests for $template_dir"
+    go run cmd/abc/abc.go templates golden-test verify $template_dir
```
Comment (sethvargo) [inline run_golden_tests.sh:11]: I wonder if this is an indication that we should have a built-in flag like `-recurse` and have the `golden-test` command handle the nesting?

Comment (drevell) [reply run_golden_tests.sh:11]: I agree. We have #311 for that, coming soon.

**Review by sethvargo — APPROVED** [review]
LGTM, but CI is unhap.

**Review by sethvargo — APPROVED** [review]

**Review by mattrandallbecker — APPROVED** [review]
+100 to shift left for CI failures

**Pull request merged**